### PR TITLE
Extend lifecycle protos to query all approved chaincode definitions

### PR DESCRIPTION
**Pull request by satota2**
This patch extends lifecycle protos to query all approved chaincode definitions.

See https://github.com/hyperledger/fabric/issues/4564 for the detail.

#### Type of change

- New feature

#### Description

As a sub-task stemming from issue #4564, this patch extends lifecycle protos to query all approved chaincode definitions.
I have already done the provisional implementation and will submit them to the Fabric repository once this patch is merged.

The example of output when executing the enhanced queryapproved in the provisional implementation:

```
$ peer lifecycle chaincode queryapproved -C mychannel
Approved chaincode definition on channel 'mychannel':
name: basic2, sequence: 2, version: 1.0.1, init-required: false, package-id: basic2_1.0.1:dae4dca432d56265e87e6416b602b40e94e7f7cdc177031abda1c81d9ed4258a, endorsement plugin: escc, validation plugin: vscc
name: basic, sequence: 1, version: 1.0.1, init-required: false, package-id: basic_1.0.1:f4babb5fd92c0ab4bce8c6ac30ca7bbb4a55e6c37774582d11639b6036ae0273, endorsement plugin: escc, validation plugin: vscc
name: basic2, sequence: 1, version: 1.0.1, init-required: false, package-id: basic2_1.0.1:dae4dca432d56265e87e6416b602b40e94e7f7cdc177031abda1c81d9ed4258a, endorsement plugin: escc, validation plugin: vscc
```

```
 $ peer lifecycle chaincode queryapproved -C mychannel -O json
{
        "approved_chaincode_definitions": [
                {
                        "name": "basic2",
                        "sequence": 2,
                        "version": "1.0.1",
                        "endorsement_plugin": "escc",
                        "validation_plugin": "vscc",
                        "validation_parameter": "EiAvQ2hhbm5lbC9BcHBsaWNhdGlvbi9FbmRvcnNlbWVudA==",
                        "collections": {},
                        "source": {
                                "Type": {
                                        "LocalPackage": {
                                                "package_id": "basic2_1.0.1:dae4dca432d56265e87e6416b602b40e94e7f7cdc177031abda1c81d9ed4258a"
                                        }
                                }
                        }
                },
                {
                        "name": "basic",
                        "sequence": 1,
                        "version": "1.0.1",
                        "endorsement_plugin": "escc",
                        "validation_plugin": "vscc",
                        "validation_parameter": "EiAvQ2hhbm5lbC9BcHBsaWNhdGlvbi9FbmRvcnNlbWVudA==",
                        "collections": {},
                        "source": {
                                "Type": {
                                        "LocalPackage": {
                                                "package_id": "basic_1.0.1:f4babb5fd92c0ab4bce8c6ac30ca7bbb4a55e6c37774582d11639b6036ae0273"
                                        }
                                }
                        }
                },
                {
                        "name": "basic2",
                        "sequence": 1,
                        "version": "1.0.1",
                        "endorsement_plugin": "escc",
                        "validation_plugin": "vscc",
                        "validation_parameter": "EiAvQ2hhbm5lbC9BcHBsaWNhdGlvbi9FbmRvcnNlbWVudA==",
                        "collections": {},
                        "source": {
                                "Type": {
                                        "LocalPackage": {
                                                "package_id": "basic2_1.0.1:dae4dca432d56265e87e6416b602b40e94e7f7cdc177031abda1c81d9ed4258a"
                                        }
                                }
                        }
                }
        ]
}
```

### CODE REVIEW

File: peer/lifecycle/lifecycle.proto [hunk 1]
```diff
@@ -188,6 +188,30 @@ message QueryApprovedChaincodeDefinitionResult {
     ChaincodeSource source = 8;
 }
 
+// QueryApprovedChaincodeDefinitionsArgs is the message used as arguments to
+// `_lifecycle.QueryApprovedChaincodeDefinitions`.
+message QueryApprovedChaincodeDefinitionsArgs {
+    string name = 1;
+    int64 sequence = 2;
```
Comment (denyeart) [inline peer/lifecycle/lifecycle.proto:195]: I'm confused, if the new function doesn't require client to pass in name and sequence, why are they listed as input parameters here?

Comment (satota2) [reply peer/lifecycle/lifecycle.proto]: @denyeart Thank you for your review. I apologize for the oversight; I was aware of the need to remove these parameters but inadvertently missed updating them in the final submission. You are correct that these input parameters are not necessary for this function. I have now corrected this and resubmitted the patch. Please take a moment to review the updated version.

**Review by denyeart — APPROVED** [review]

**Comment by satota2** [comment]
@denyeart 
Thank you for merging the changes. It seems all the bindings were successfully built with the proto change (node, java, go, go apiv2). I would greatly appreciate it if you could proceed with creating a v0.3.2 PR and then generate a v0.3.2 release tag. This would allow the fabric main branch to pull in github.com/hyperledger/fabric-protos-go, helping me move forward with the implementation in the fabric main repository. Thank you in advance.

**Comment by denyeart** [comment]
@satota2 v0.3.2 released!

**Pull request merged**